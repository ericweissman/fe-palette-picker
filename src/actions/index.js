export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
});

export const hasErrored = (message) => ({
  type: 'HAS_ERRORED',
  message
});

//projects
export const getProjectsSuccess = (projects) => ({
  type: 'GET_PROJECTS_SUCCESS',
  projects
})

export const addProjectSuccess = (project) => ({
  type: 'ADD_PROJECT_SUCCESS',
  project
})