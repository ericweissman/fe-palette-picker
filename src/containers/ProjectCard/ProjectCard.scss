.project-card {
  border: 2px solid seagreen;
  margin: 10px;
}