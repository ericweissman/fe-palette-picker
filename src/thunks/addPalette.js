import { isLoading, hasErrored, addPaletteSuccess } from '../actions'

export const addPalette = (url, palette) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: "POST",
        headers: {
          "Content-Type": "application/json"
        },
        body: JSON.stringify(palette)
      })
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const result = await response.json()
      dispatch(addPaletteSuccess(result))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}