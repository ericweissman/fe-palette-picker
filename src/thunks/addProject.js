import { isLoading, hasErrored, addProjectSuccess } from '../actions'

export const addProject = (url, project_name) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: "POST",
        headers: {
          "Content-Type" : "application/json"
        },
        body: JSON.stringify({ project_name })
      })
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const result = await response.json()
      dispatch(addProjectSuccess(result))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}