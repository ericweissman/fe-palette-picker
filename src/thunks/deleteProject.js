import { isLoading, hasErrored, deleteProjectSuccess } from '../actions'

export const deleteProject = (url, projectID) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: "DELETE",
        headers: {
          "Content-Type" : "application/json"
        }
      })
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      // const result = await response.json()
      dispatch(deleteProjectSuccess(projectID))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}