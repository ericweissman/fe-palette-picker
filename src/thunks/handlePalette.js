import { isLoading, hasErrored } from '../actions'

export const handlePalette = (url, actionToDispatch, method, palette) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: `${method}`,
        headers: {
          "Content-Type": "application/json"
        },
        body: JSON.stringify(palette)
      })
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const result = await response.json()
      dispatch(actionToDispatch(result))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}