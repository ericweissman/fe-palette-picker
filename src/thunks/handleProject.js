import { isLoading, hasErrored } from '../actions'

export const handleProject = (url, actionToDispatch, method, project) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: `${method}`,
        headers: {
          "Content-Type": "application/json"
        },
        body: JSON.stringify(project)
      })
      if (!response.ok) {
        throw Error(response.statusText)
      }
      
      dispatch(isLoading(false))
      const result = await response.json()
      dispatch(actionToDispatch(result))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}