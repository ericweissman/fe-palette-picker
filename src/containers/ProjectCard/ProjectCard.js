import React, { Component } from 'react';
import { connect } from 'react-redux'
import { deleteProjectSuccess, deletePaletteSuccess, setActivePalette, getPalettesSuccess, editProjectSuccess, setPaletteToEdit } from '../../actions';
import { handleProject } from '../../thunks/handleProject'
import { handlePalette } from '../../thunks/handlePalette';
import Palette from '../Palette/Palette'


class ProjectCard extends Component {
  state = {
    edited: false,
    projectName: '',
  }

  setActive = (palette) => {
    this.props.setActivePalette(palette)
  }

  deleteProject = () => {
    const { id } = this.props.project
    const project = { id }
    const url = process.env.REACT_APP_BACKEND_URL + `/api/v1/projects/${id}`
    this.props.handleProject(url, deleteProjectSuccess, 'DELETE', project)
  }

  deletePalette = (pID) => {
    const projID = this.props.project.id
    const palette = { pID }
    const url = process.env.REACT_APP_BACKEND_URL + `/api/v1/projects/${projID}/palettes/${pID}`
    this.props.handlePalette(url, deletePaletteSuccess, 'DELETE', palette)
  }

  getPalettes = () => {
    const { id } = this.props.project
    const url = process.env.REACT_APP_BACKEND_URL + `/api/v1/projects/${id}/palettes`
    this.props.handlePalette(url, getPalettesSuccess, "GET")
  }

  toggleEdited = () => {
    const { id } = this.props.project
    const { projectName } = this.state
    const url = process.env.REACT_APP_BACKEND_URL + `/api/v1/projects/${id}`
    const project = { project_name: projectName }

    this.setState({ edited: !this.state.edited })
    if (this.state.edited) {
      this.props.handleProject(url, editProjectSuccess, 'PUT', project)
    }
  }
  
  editPalette = (palette) => {
    this.props.setPaletteToEdit(palette)
  }

  updateName = (e) => {
    const { value } = e.target
    this.setState({
      projectName: value
    })
  }

  render() {
    const { project_name, id } = this.props.project
    const palettes = this.props.palettes.filter(palette => palette.project_id === id)
    const palettesToDisplay = palettes.map(palette => {
      return <Palette key={palette.id} palette={palette} setActive={this.setActive} deletePalette={this.deletePalette} editPalette={this.editPalette} editName={this.props.handlePalette} />
    })
    const { edited } = this.state

    return (
      <div className="project-card">
        <div className="project-name">
          {edited ?
            <input onChange={this.updateName} value={this.state.projectName}></input> :
            <h3>{project_name}</h3>
          }
          <button onClick={this.toggleEdited} className={edited ? "save-name-btn" : "edit-name-btn" }></button>
          <button className="delete-name-btn" onClick={this.deleteProject}></button>
        </div>
        <button className="get-palettes-btn" onClick={this.getPalettes} id={id}>Get Palettes</button>
        {palettesToDisplay}
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  projects: state.projects,
  palettes: state.palettes,
});

export const mapDispatchToProps = (dispatch) => ({
  handleProject: (url, actionToDispatch, method, project) => dispatch(handleProject(url, actionToDispatch, method, project)),
  handlePalette: (url, actionToDispatch, method, palette) => dispatch(handlePalette(url, actionToDispatch, method, palette)),
  setActivePalette: (palette) => dispatch(setActivePalette(palette)),
  setPaletteToEdit: (palette) => dispatch(setPaletteToEdit(palette))
})

export default connect(mapStateToProps, mapDispatchToProps)(ProjectCard);