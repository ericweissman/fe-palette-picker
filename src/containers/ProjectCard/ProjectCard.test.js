import React from 'react';
import { shallow } from 'enzyme';
import { handlePalette } from '../../thunks/handlePalette'
import { handleProject } from '../../thunks/handleProject'
import { ProjectCard, mapStateToProps, mapDispatchToProps } from './ProjectCard'
import { addProjectSuccess, addPaletteSuccess, setActivePalette, setPaletteToEdit } from '../../actions'

jest.mock('../../thunks/handlePalette')
jest.mock('../../thunks/handleProject')

describe('ProjectCard', () => {
  let wrapper

  describe('snapshots and initial state', () => {

    it('should have an initial state', () => {
      const mockProject = {}
      const mockPalette = [{}]

      const expected = { edited: false, projectName: '' }
      wrapper = shallow(
        <ProjectCard project={mockProject} palettes={mockPalette} />
      )

      expect(wrapper.state()).toEqual(expected)
    })

    it('should match the initial snapshot', () => {
      const mockProject = {}
      const mockPalette = [{}]

      wrapper = shallow(
        <ProjectCard project={mockProject} palettes={mockPalette} />
      )

      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot with paletteToEdit and activePalette', () => {
      const mockProject = { id: 12, project_name: 'best ever' }
      const mockPalette = [{ id: 3, color_1: '111111', color_2: '222222', color_3: '333333', color_4: '444444', color_5: '555555' }]

      wrapper = shallow(
        <ProjectCard project={mockProject} palettes={mockPalette} />
      )

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('mapStateToProps', () => {
    it('should return an object with a paletteToEdit object and activePalette array', () => {
      const mockState = { projects: [{}, {}], palettes: [{}, {}, {}, {}, {}], activePalette: [] }
      const expected = { projects: [{}, {}], palettes: [{}, {}, {}, {}, {}] }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call handleProject dispatch', () => {
      const mockDispatch = jest.fn()
      const mockURL = 'http://localhost:3001/api/v1/projects/1/palettes'
      const mockProject = {}

      const actionToDispatch = handleProject(mockURL, setActivePalette, 'POST', mockProject)

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.handleProject(mockURL, addProjectSuccess, 'POST', mockProject)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should call handlePalette dispatch', () => {
      const mockDispatch = jest.fn()
      const mockURL = 'http://localhost:3001/api/v1/projects/1/palettes'
      const mockPalette = {}

      const actionToDispatch = handlePalette(mockURL, setActivePalette, 'POST', mockPalette)

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.handlePalette(mockURL, addPaletteSuccess, 'POST', mockPalette)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should call setActivePalette dispatch', () => {
      const mockDispatch = jest.fn()
      const mockPalette = { id: 4, palette_name: 'palette 1', color_1: '111111', color_2: '222222', color_3: '333333', color_4: '444444', color_5: '555555' }

      const actionToDispatch = setActivePalette(mockPalette)

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setActivePalette(mockPalette)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should call setPaletteToEdit dispatch', () => {
      const mockDispatch = jest.fn()
      const mockPalette = { id: 4, palette_name: 'palette 1', color_1: '111111', color_2: '222222', color_3: '333333', color_4: '444444', color_5: '555555' }

      const actionToDispatch = setPaletteToEdit(mockPalette)

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setPaletteToEdit(mockPalette)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })

  describe('setActive', () => {
    it('should set an active palette', () => {
      const mockProject = {}
      const mockPalette = [{}]
      const mockActivePalette = jest.fn()
      const mockProps = {
        project: mockProject,
        palettes: mockPalette,
        setActivePalette: mockActivePalette
      }

      wrapper = shallow(
        <ProjectCard {...mockProps} />
      )
      wrapper.instance().setActive()

      expect(mockProps.setActivePalette).toHaveBeenCalled()
    })
  })

  describe('editPalette', () => {
    it('should set a palette to edit', () => {
      const mockProject = {}
      const mockPalette = [{}]
      const mockPaletteToEdit = jest.fn()
      const mockProps = {
        project: mockProject,
        palettes: mockPalette,
        setPaletteToEdit: mockPaletteToEdit
      }

      wrapper = shallow(
        <ProjectCard {...mockProps} />
      )
      wrapper.instance().editPalette()

      expect(mockProps.setPaletteToEdit).toHaveBeenCalled()
    })
  })

  describe('updateName', () => {
    it('should update the projectName', () => {
      const mockProject = {}
      const mockPalette = [{}]
      const mockPaletteToEdit = jest.fn()
      const mockProps = {
        project: mockProject,
        palettes: mockPalette,
        setPaletteToEdit: mockPaletteToEdit
      }
      const mockEvent = { target: { value: 'z' } }
      const expected = { edited: false, projectName: 'z' }

      wrapper = shallow(
        <ProjectCard {...mockProps} />
      )
      wrapper.instance().updateName(mockEvent)

      expect(wrapper.state()).toEqual(expected)
    })
  })
})