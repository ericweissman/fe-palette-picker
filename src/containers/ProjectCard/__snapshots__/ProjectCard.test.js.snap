// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard snapshots and initial state should match the initial snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProjectCard
    palettes={
      Array [
        Object {},
      ]
    }
    project={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <h3 />
          <button
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            onClick={[Function]}
          >
            Delete
          </button>
        </div>,
        <button
          onClick={[Function]}
        >
          Get Palettes
        </button>,
        Array [
          <Connect(Palette)
            deletePalette={[Function]}
            editPalette={[Function]}
            palette={Object {}}
            setActive={[Function]}
          />,
        ],
      ],
      "className": "project-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3 />,
            <button
              onClick={[Function]}
            >
              Edit
            </button>,
            <button
              onClick={[Function]}
            >
              Delete
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Edit",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Edit",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Delete",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Delete",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Get Palettes",
          "id": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Get Palettes",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "deletePalette": [Function],
          "editName": undefined,
          "editPalette": [Function],
          "palette": Object {},
          "setActive": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <h3 />
            <button
              onClick={[Function]}
            >
              Edit
            </button>
            <button
              onClick={[Function]}
            >
              Delete
            </button>
          </div>,
          <button
            onClick={[Function]}
          >
            Get Palettes
          </button>,
          Array [
            <Connect(Palette)
              deletePalette={[Function]}
              editPalette={[Function]}
              palette={Object {}}
              setActive={[Function]}
            />,
          ],
        ],
        "className": "project-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3 />,
              <button
                onClick={[Function]}
              >
                Edit
              </button>,
              <button
                onClick={[Function]}
              >
                Delete
              </button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Edit",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Edit",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Delete",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Delete",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Get Palettes",
            "id": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Get Palettes",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "deletePalette": [Function],
            "editName": undefined,
            "editPalette": [Function],
            "palette": Object {},
            "setActive": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`ProjectCard snapshots and initial state should match the snapshot with paletteToEdit and activePalette 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProjectCard
    palettes={
      Array [
        Object {
          "color_1": "111111",
          "color_2": "222222",
          "color_3": "333333",
          "color_4": "444444",
          "color_5": "555555",
          "id": 3,
        },
      ]
    }
    project={
      Object {
        "id": 12,
        "project_name": "best ever",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <h3>
            best ever
          </h3>
          <button
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            onClick={[Function]}
          >
            Delete
          </button>
        </div>,
        <button
          id={12}
          onClick={[Function]}
        >
          Get Palettes
        </button>,
        Array [],
      ],
      "className": "project-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              best ever
            </h3>,
            <button
              onClick={[Function]}
            >
              Edit
            </button>,
            <button
              onClick={[Function]}
            >
              Delete
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "best ever",
            },
            "ref": null,
            "rendered": "best ever",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Edit",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Edit",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Delete",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Delete",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Get Palettes",
          "id": 12,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Get Palettes",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <h3>
              best ever
            </h3>
            <button
              onClick={[Function]}
            >
              Edit
            </button>
            <button
              onClick={[Function]}
            >
              Delete
            </button>
          </div>,
          <button
            id={12}
            onClick={[Function]}
          >
            Get Palettes
          </button>,
          Array [],
        ],
        "className": "project-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                best ever
              </h3>,
              <button
                onClick={[Function]}
              >
                Edit
              </button>,
              <button
                onClick={[Function]}
              >
                Delete
              </button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "best ever",
              },
              "ref": null,
              "rendered": "best ever",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Edit",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Edit",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Delete",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Delete",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Get Palettes",
            "id": 12,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Get Palettes",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
